client/dialogs/dialog-type.ts

```diff
@@ -20,3 +20,125 @@ export enum DialogType {
   TermsOfService = 'termsOfService',
   Whispers = 'whispers',
 }
+
+export interface BaseDialogPayload {
+  type: DialogType
+  initData?: Record<string, unknown>
```

Comment:
honestly feels weird to have this, I think you could just not include this an a type union would work out fine? And once you remove this I'm not sure having this base type is really worthwhile either, just feels like a lot of extra code boilerplate for something that will be enforced by the code using it anyway

Reply:
Hmm, I'm not sure if I'm understanding TypeScript correctly, but I thought that we need _something_ to discriminate `initData` on for different Dialog types. And I thought my only options to do that were to either add `dialogType` property to the `initData` itself, or to type the whole payload like this since it already should have dialog type.

Reply:
The only reason I can see this base type being useful is for use in a reducer that has no knowledge of the underlying dialog type. I didn't think about that before, so I guess it can stay if that's your reason for using it. (But no, this is not a necessary thing to have for TS to figure out what the type of `initData` is, TS will use whatever unique properties exist in the types in the union to figure out which specific type you are referring to)

If it is not possible to have any other fields (and that seems to be the case, to me), then you'd probably be better off declaring this as a generic type with 2 parameters: a DialogType and an `initData` type, and just using that for all the payloads. That would at least be properly restrictive.